crates/kitsune_p2p/types/src/config.rs

```diff
@@ -333,3 +339,324 @@ pub mod tuning_params_struct {
+    #[allow(dead_code)] // because of feature flipping
+    pub fn is_tx2(&self) -> bool {
+        #[cfg(feature = "tx2")]
+        {
+            #[cfg(feature = "tx5")]
+            {
+                if let Some(t) = self.transport_pool.get(0) {
+                    !matches!(t, TransportConfig::WebRTC { .. })
+                } else {
+                    true
+                }
+            }
+            #[cfg(not(feature = "tx5"))]
+            {
+                true
+            }
+        }
+        #[cfg(not(feature = "tx2"))]
+        {
+            false
+        }
+    }
+
+    /// This config is making use of tx5 transport
+    #[allow(dead_code)] // because of feature flipping
+    pub fn is_tx5(&self) -> bool {
+        #[cfg(feature = "tx5")]
+        {
+            if let Some(t) = self.transport_pool.get(0) {
+                return matches!(t, TransportConfig::WebRTC { .. });
+            }
+        }
+        false
+    }
+
+    /// `tx2` is currently designed to use exactly one proxy wrapped transport,
+    /// so convert a bunch of the options from the previous transport
+    /// paradigm into that pattern.
+    #[cfg(feature = "tx2")]
+    pub fn to_tx2(&self) -> KitsuneResult<KitsuneP2pTx2Config> {
+        use KitsuneP2pTx2ProxyConfig::*;
+        match self.transport_pool.get(0) {
+            /*
+            Some(TransportConfig::Proxy {
+                sub_transport,
+                proxy_config,
+            }) => {
+                let backend = match &**sub_transport {
+                    TransportConfig::Mem {} => KitsuneP2pTx2Backend::Mem,
+                    TransportConfig::Quic { bind_to, .. } => {
+                        let bind_to = cnv_bind_to(bind_to);
+                        KitsuneP2pTx2Backend::Quic { bind_to }
+                    }
+                    _ => return Err("kitsune tx2 backend must be mem or quic".into()),
+                };
+                let use_proxy = match proxy_config {
+                    ProxyConfig::RemoteProxyClient { proxy_url } => {
+                        Specific(proxy_url.clone().into())
+                    }
+                    ProxyConfig::RemoteProxyClientFromBootstrap {
+                        bootstrap_url,
+                        fallback_proxy_url,
+                    } => Bootstrap {
+                        bootstrap_url: bootstrap_url.clone().into(),
+                        fallback_proxy_url: fallback_proxy_url.clone().map(Into::into),
+                    },
+                    ProxyConfig::LocalProxyServer { .. } => NoProxy,
+                };
+                Ok(KitsuneP2pTx2Config { backend, use_proxy })
+            }
+            Some(TransportConfig::Quic { bind_to, .. }) => {
+                let bind_to = cnv_bind_to(bind_to);
+                Ok(KitsuneP2pTx2Config {
+                    backend: KitsuneP2pTx2Backend::Quic { bind_to },
+                    use_proxy: NoProxy,
+                })
+            }
+            */
```

Comment:
Do we still need all these commented lines, as tx2 is being phased out?

Reply:
We don't https://github.com/holochain/holochain/pull/2689/files#diff-1a8dd559c02e493cbcfa088e8da572bbe9ac95fc2ebcde414ccd942eab78661aL126-L161

Reply:
deleted